src/Symfony/Component/Console/Exception/UnknownCommandException.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Exception;
+
+/**
+ * @author Martin Hasoň <martin.hason@gmail.com>
+ */
+class UnknownCommandException extends CommandNotFoundException
+{
+    private $command;
+
+    public function __construct($command, $alternatives = array(), $code = null, $previous = null)
+    {
+        $this->command = $command;
+
+        $message = sprintf('Command "%s" is not defined.', $command);
+
+        if ($alternatives) {
+            if (1 == count($alternatives)) {
+                $message .= "\n\nDid you mean this?\n    ";
```

Comment:
I'm not sure the exception should format its message itself. Instead, consumers should build the formatted message IMO.